library/core/src/intrinsics.rs

```diff
@@ -1846,20 +1847,22 @@ pub(crate) fn is_nonoverlapping<T>(src: *const T, dst: *const T, count: usize) -
 /// [`Vec::append`]: ../../std/vec/struct.Vec.html#method.append
 #[doc(alias = "memcpy")]
 #[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_const_unstable(feature = "const_intrinsic_copy", issue = "none")]
 #[inline]
-pub unsafe fn copy_nonoverlapping<T>(src: *const T, dst: *mut T, count: usize) {
+pub const unsafe fn copy_nonoverlapping<T>(src: *const T, dst: *mut T, count: usize) {
     extern "rust-intrinsic" {
         fn copy_nonoverlapping<T>(src: *const T, dst: *mut T, count: usize);
     }
 
-    if cfg!(debug_assertions)
+    // FIXME: Perform these cheks in a const compatible way?
```

Comment:
The goal isn't to perform the checks at const-time, the goal is to find a way to only perform them when this code is executed at runtime.

Reply:
So something like

```suggestion
    // FIXME: Perform these checks only at run time
```
?

Reply:
Yeah that sounds good. :)